dev/run-tests.py

```diff
@@ -371,9 +371,10 @@ def build_apache_spark(build_tool, hadoop_version):
         build_spark_sbt(hadoop_version)
 
 
-def detect_binary_inop_with_mima():
+def detect_binary_inop_with_mima(hadoop_version):
+    build_profiles = get_hadoop_profiles(hadoop_version) + modules.root.build_profile_flags
     set_title_and_block("Detecting binary incompatibilities with MiMa", "BLOCK_MIMA")
-    run_cmd([os.path.join(SPARK_HOME, "dev", "mima")])
+    run_cmd([os.path.join(SPARK_HOME, "dev", "mima")] + build_profiles)
```

Comment:
I added these arguments so that when the mima script would end up re-running the packaging step, it would not copy new files with different versions into the assembly `jars` directory. That was causing test failures earlier if you look at the comment history.

Later I added a flag to disable copying dependencies altogether, and used it in the mima invocation, making this not entirely necessary. But in the end, I think this is still more correct than the previous code - and, as you say, has the nice effect of making sure binary compatibility doesn't change because of different dependency versions.